XmlFormatterOsIndependent/ViewModels/ViewModelBase.cs

```diff
@@ -0,0 +1,146 @@
+﻿using PluginFramework.Interfaces.Manager;
+using ReactiveUI;
+using System;
+using System.Runtime.InteropServices;
+using System.Windows.Input;
+using XmlFormatterModel.Setting;
+using XmlFormatterOsIndependent.Commands;
+using XmlFormatterOsIndependent.DataSets;
+using XmlFormatterOsIndependent.Enums;
+
+namespace XmlFormatterOsIndependent.ViewModels
+{
+    /// <summary>
+    /// Base view model class
+    /// </summary>
+    public class ViewModelBase : ReactiveObject
+    {
+        /// <summary>
+        /// The current view of this model
+        /// </summary>
+        protected readonly ViewContainer view;
+
+        /// <summary>
+        /// The settings manager to use
+        /// </summary>
+        protected readonly ISettingsManager settingsManager;
+
+        /// <summary>
+        /// The plugin manager to use
+        /// </summary>
+        protected readonly IPluginManager pluginManager;
+
+        /// <summary>
+        /// The path to the settings file
+        /// </summary>
+        protected readonly string settingsPath;
+
+        /// <summary>
+        /// Create a new instance of this class
+        /// </summary>
+        public ViewModelBase() : this(null, null, null)
+        {
+
+        }
+
+        /// <summary>
+        /// Create a new instance of this class
+        /// </summary>
+        /// <param name="view">The view for this model</param>
+        /// <param name="settingsManager">The settings manager for this model</param>
+        /// <param name="pluginManager">The plugin manager for this model</param>
+        public ViewModelBase(ViewContainer view, ISettingsManager settingsManager, IPluginManager pluginManager)
+        {
+            this.view = view;
+            this.settingsManager = settingsManager;
+            this.pluginManager = pluginManager;
+            settingsPath = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+            settingsPath += System.IO.Path.DirectorySeparatorChar + "XmlFormatter";
+            settingsPath += System.IO.Path.DirectorySeparatorChar + "settings.set";
+
+            settingsManager.Load(settingsPath);
+            ChangeTheme();
+            DoOSSpecific();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this call from a constructor to the overridable 'DoOSSpecific' method.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5634024)